src/lightning_app/utilities/cluster_logs.py

```diff
@@ -0,0 +1,108 @@
+import json
+import queue
+from dataclasses import dataclass
+from threading import Thread
+from typing import Callable, Iterator, Optional
+
+import arrow
+import dateutil.parser
+from websocket import WebSocketApp
+
+from lightning_app.utilities.log_helpers import _error_callback, _OrderedLogEntry
+from lightning_app.utilities.logs_socket_api import _ClusterLogsSocketAPI
+from lightning_app.utilities.network import LightningClient
+
+
+@dataclass
+class _ClusterLogEventLabels:
+    cluster_id: str
+    grid_url: str
+    hostname: str
+    level: str
+    logger: str
+    path: Optional[str] = None
+    workspace: Optional[str] = None
+    identifier: Optional[str] = None
+    issuer: Optional[str] = None
+    error: Optional[str] = None
+    errorVerbose: Optional[str] = None
+
+
+@dataclass
+class _ClusterLogEvent(_OrderedLogEntry):
+    labels: _ClusterLogEventLabels
+
+
+def _push_log_events_to_read_queue_callback(read_queue: queue.PriorityQueue):
+    """Pushes _LogEvents from websocket to read_queue.
+
+    Returns callback function used with `on_message_callback` of websocket.WebSocketApp.
+    """
+
+    def callback(ws_app: WebSocketApp, msg: str):
+        # We strongly trust that the contract on API will hold atm :D
+        event_dict = json.loads(msg)
+        labels = _ClusterLogEventLabels(**event_dict["labels"])
+
+        if "message" in event_dict:
+            message = event_dict["message"]
+            timestamp = dateutil.parser.isoparse(event_dict["timestamp"])
+            event = _ClusterLogEvent(
+                message=message,
+                timestamp=timestamp,
+                labels=labels,
+            )
+            read_queue.put(event)
```

Comment:
We can (if I remember correctly) push tuple (ts, event) and queue will take weight from first argument (ts) 
So there will be no need to implement < and <= operators for `_ClusterLogEvent` :)
Also inheritance on those Event classes seems odd (code smell)

Reply:
Sure, nice idea! I think yes, but maybe it should be unified with app logs :) Maybe can be a subject of a subsequent PR to don't touch too many things here.

Reply:
👍